### PR TITLE
Add CI Actions

### PR DESCRIPTION
**Pull request by emizzle**

### CODE REVIEW

**Comment by emizzle** [comment]
This should be g2g now.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+            cpu: amd64
+          - os: linux
+            cpu: i386
+          - os: macos
+            cpu: amd64
+          - os: windows
+            cpu: amd64
+          #- os: windows
+            #cpu: i386
+        branch: [version-1-2, version-1-4, version-1-6, devel]
```
Comment (cskiraly) [inline .github/workflows/ci.yml:26]: is this normal to have 'devel' in the matrix? It feels like a good early warning, but we should treat it as a warning only, right?

File: tests/dht/test_providers.nim [hunk 2]
```diff
@@ -166,54 +166,55 @@ suite "Providers Tests: two nodes":
 
 
 
-suite "Providers Tests: 20 nodes":
+# SKIP: currently this is not working even with a 2 minute timeout
+# suite "Providers Tests: 20 nodes":
```
Comment (cskiraly) [inline tests/dht/test_providers.nim:170]: I think the only one potentially failing is the last test of the suite, the one killing the bootstrap node. Maybe we should keep the others.

**Comment by cskiraly** [comment]
As mentioned in detailed comments, I would suggest to re-enable the tests except the last one (if I got well what fails), and then you can merge.

**Comment by emizzle** [comment]
> As mentioned in detailed comments, I would suggest to re-enable the tests except the last one (if I got well what fails), and then you can merge.

Done.

**Comment by cskiraly** [comment]
The test failure is really strange and it is only in "CI / linux-i386 (Nim version-1-2) (pull_request) Failing after 7m — linux-i386 (Nim version-1-2)". I would merge anyway.

**Comment by emizzle** [comment]
After re-running the tests, it is failing on more platforms. Error is with the second-to-last 20 node test:
```
DBG 2022-03-06 23:04:17.122+00:00 ---- STARTING CHECKS ---                   tid=11989
    /home/runner/work/nim-libp2p-dht/nim-libp2p-dht/tests/dht/test_providers.nim(219, 10): Check failed: (providers.len == 1 and providers[0].peerId == nodes[0].toPeerRecord.peerId)
  [FAILED] 20 nodes, retieve after bootnode dies
```
https://github.com/status-im/nim-libp2p-dht/runs/5441203841?check_suite_focus=true#step:11:2283

**Comment by cskiraly** [comment]
I've just seen skip() is not doing what one would expect. I add a patch that removes skip() and comments out the section.

**Pull request merged**